packages/design-system/src/components/web-components/ds-drawer/ds-drawer.test.tsx

```diff
@@ -0,0 +1,122 @@
+import { render, screen, cleanup } from '@testing-library/preact';
+import userEvent from '@testing-library/user-event';
+import './ds-drawer';
+
+const children = (
+  <>
+    <strong>An Explanation</strong>
+    <p>
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
+      labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
+      laboris nisi ut aliquip ex ea commodo consequat.
+    </p>
+  </>
+);
+
+function renderDrawer(args, children) {
+  return render(
+    <ds-drawer {...(args as any)}>
+      {children}
+      {args['footer-body'] && (
```

Comment:
I don't think you need to check for the attribute and then put in the slot. See above comment.